app/frontend/src/search/ui/sort.js

```diff
@@ -22,4 +22,10 @@ export const renderSortSelectInput = (renderOptions, isFirstRender) => {
   }
 };
 
-export const getSearchReplicaName = (selected) => (selected ? `Vacancy_${selected}` : 'Vacancy');
+export const getSearchIndexName = (selected) => {
```

Comment:
You don't need to change this function at all. You do need to add a function that sets the appropriate default sort based on user input.